docs/BUILD_AND_TEST.md

```diff
@@ -83,8 +76,22 @@ It should be noted here that if there are any untagged tests in the directory re
 
 Once the maintainers feel that the *small* tests provide sufficient code coverage, the existing *legacy* tests will be phased out (or used in the construction of a set of *medium* and *large* tests for the Snap CI/CD toolchain). All new tests being added to the Snap framework by contributors should be marked as either `small`, `medium`, or `large` tests, depending on their scope.
 
+### Building Effective Small Tests
```

Comment:
In the longer run, yes. Out of scope for this PR but wonder if @nanliu or @kindermoumoute feels that should be opened as a new issue.